lib/ansible/module_utils/facts/system/date_time.py

```diff
@@ -32,7 +36,10 @@ def collect(self, module=None, collected_facts=None):
         facts_dict = {}
         date_time_facts = {}
 
-        now = datetime.datetime.now()
+        utcnow = datetime.utcnow()
+        offset_delta = timedelta(seconds=time.timezone)
```

Comment:
I might recommend using `time.time()` as the base, and then calculate local and utc using:

```
ts = time.time()
now = datetime.datetime.fromtimestamp(ts)
utcnow = datetime.datetime.utcfromtimestamp(ts)
```

Reply:
That’s a great suggestion. Thanks! I went through several different approaches (and spent way too much time reading docs) but I didn’t try using `time.time()`. I’ll give it a shot.

Reply:
Why does `datetime.utcnow()` not work here?

Reply:
The problem is getting both utc and local time, that match.  Calling both gives you different datetime objects. To convert between utc and local, you have to know the offset, potentially including DST to calculate it.  Going the `time.time` route, and creating a datetime from the timestamp resolves that complexity.

Reply:
Ah, right, forgot that it's PITA in Python 2... Python 3 is easier:
```python
$ ipython
Python 3.7.1 (default, Jan 28 2019, 08:25:13) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from datetime import datetime

In [2]: utcn = datetime.utcnow()

In [3]: utcn
Out[3]: datetime.datetime(2020, 7, 16, 21, 43, 35, 56745)

In [4]: utcn.astimezone()
Out[4]: datetime.datetime(2020, 7, 16, 21, 43, 35, 56745, tzinfo=datetime.time
zone(datetime.timedelta(seconds=7200), 'CEST'))
```

Reply:
After digging into this far too much, I think I'm going to change the testing approach to validate the calls to `datetime.datetime.strftime()` since mocking out `time` is far from trivial.